packages/venia-concept/src/components/FilterModal/FilterList/filterList.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import classify from 'src/classify';
+import defaultClasses from './filterList.css';
+import { List } from '@magento/peregrine';
+import FilterDefault from './filterDefault';
+import FilterSwatch from './filterSwatch';
+import { WithFilterSearch } from 'src/components/FilterModal/FilterSearch';
+
+class FilterList extends Component {
+    static propTypes = {
+        classes: PropTypes.shape({
+            filterItem: PropTypes.string
+        }),
+        chosenOptions: PropTypes.arrayOf(
+            PropTypes.shape({
+                title: PropTypes.string,
+                value: PropTypes.string
+            })
+        ),
+        layoutClass: PropTypes.string,
+        isSwatch: PropTypes.bool,
+        options: PropTypes.object,
+        filterAdd: PropTypes.func,
+        filterRemove: PropTypes.func,
+        items: PropTypes.array
+    };
+
+    toggleOption = event => {
+        const { filterRemove, filterAdd } = this.props;
+        const { value, title, dataset } =
+            event.currentTarget || event.srcElement;
+        const { group } = dataset;
+        const item = { title, value, group };
+        this.isOptionActive(item) ? filterRemove(item) : filterAdd(item);
+    };
+
+    isOptionActive = option =>
+        this.props.chosenOptions.findIndex(
+            item => item.value === option.value && item.name === option.name
+        ) > -1;
+
+    isFilterSelected = item => {
+        return !!this.props.chosenOptions.find(
+            ({ title, value }) =>
+                item.label === title && item.value_string === value
+        );
+    };
+
+    render() {
+        const { toggleOption, isFilterSelected } = this;
+        const {
+            classes,
+            items,
+            id,
+            options,
+            layoutClass,
+            isSwatch
+        } = this.props;
+
+        return (
+            <List
+                items={items}
+                getItemKey={({ value_string }) => `item-${id}-${value_string}`}
+                render={props => (
+                    <ul className={layoutClass}>{props.children}</ul>
+                )}
+                renderItem={({ item }) => {
+                    const isActive = isFilterSelected(item);
+                    const swatchProps = {
```

Comment:
Can we rename this prop to `filterProps` since not all filter items are swatches?

Reply:
Renamed in the latest update